[2025-07-17T18:57:43.551+0000] {taskinstance.py:1956} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: spy_yfinance_parquet_ingestion.fetch_and_upload_parquet_to_s3 manual__2025-07-17T18:57:42.734617+00:00 [queued]>
[2025-07-17T18:57:43.553+0000] {taskinstance.py:1956} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: spy_yfinance_parquet_ingestion.fetch_and_upload_parquet_to_s3 manual__2025-07-17T18:57:42.734617+00:00 [queued]>
[2025-07-17T18:57:43.553+0000] {taskinstance.py:2170} INFO - Starting attempt 1 of 2
[2025-07-17T18:57:43.557+0000] {taskinstance.py:2191} INFO - Executing <Task(PythonOperator): fetch_and_upload_parquet_to_s3> on 2025-07-17 18:57:42.734617+00:00
[2025-07-17T18:57:43.560+0000] {standard_task_runner.py:60} INFO - Started process 261 to run task
[2025-07-17T18:57:43.564+0000] {standard_task_runner.py:87} INFO - Running: ['***', 'tasks', 'run', 'spy_yfinance_parquet_ingestion', 'fetch_and_upload_parquet_to_s3', 'manual__2025-07-17T18:57:42.734617+00:00', '--job-id', '20', '--raw', '--subdir', 'DAGS_FOLDER/s3_ingestion_dag.py', '--cfg-path', '/tmp/tmp66sjseqt']
[2025-07-17T18:57:43.565+0000] {standard_task_runner.py:88} INFO - Job 20: Subtask fetch_and_upload_parquet_to_s3
[2025-07-17T18:57:43.584+0000] {task_command.py:423} INFO - Running <TaskInstance: spy_yfinance_parquet_ingestion.fetch_and_upload_parquet_to_s3 manual__2025-07-17T18:57:42.734617+00:00 [running]> on host 2758917a23bd
[2025-07-17T18:57:43.612+0000] {taskinstance.py:2480} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='mark' AIRFLOW_CTX_DAG_ID='spy_yfinance_parquet_ingestion' AIRFLOW_CTX_TASK_ID='fetch_and_upload_parquet_to_s3' AIRFLOW_CTX_EXECUTION_DATE='2025-07-17T18:57:42.734617+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2025-07-17T18:57:42.734617+00:00'
[2025-07-17T18:57:43.614+0000] {s3_ingestion_dag.py:19} INFO - Fetching SPY (attempt 1/5)
[2025-07-17T18:57:44.516+0000] {multi.py:182} ERROR - 
1 Failed download:
[2025-07-17T18:57:44.517+0000] {multi.py:195} ERROR - ['SPY']: YFRateLimitError('Too Many Requests. Rate limited. Try after a while.')
[2025-07-17T18:57:44.526+0000] {taskinstance.py:2698} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/opt/airflow/dags/s3_ingestion_dag.py", line 30, in fetch_and_upload_parquet_to_s3
    raise ValueError("Yahoo returned 0 rows")
ValueError: Yahoo returned 0 rows

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/models/taskinstance.py", line 433, in _execute_task
    result = execute_callable(context=context, **execute_callable_kwargs)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/operators/python.py", line 199, in execute
    return_value = self.execute_callable()
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/operators/python.py", line 216, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
  File "/opt/airflow/dags/s3_ingestion_dag.py", line 47, in fetch_and_upload_parquet_to_s3
    raise AirflowFailException(f"Ingest failed: {e}") from e
airflow.exceptions.AirflowFailException: Ingest failed: Yahoo returned 0 rows
[2025-07-17T18:57:44.534+0000] {taskinstance.py:1138} INFO - Immediate failure requested. Marking task as FAILED. dag_id=spy_yfinance_parquet_ingestion, task_id=fetch_and_upload_parquet_to_s3, execution_date=20250717T185742, start_date=20250717T185743, end_date=20250717T185744
[2025-07-17T18:57:44.541+0000] {standard_task_runner.py:107} ERROR - Failed to execute job 20 for task fetch_and_upload_parquet_to_s3 (Ingest failed: Yahoo returned 0 rows; 261)
[2025-07-17T18:57:44.561+0000] {local_task_job_runner.py:234} INFO - Task exited with return code 1
[2025-07-17T18:57:44.572+0000] {taskinstance.py:3280} INFO - 0 downstream tasks scheduled from follow-on schedule check
